test/unittests/utils/test_dwi.py

```diff
@@ -53,3 +56,221 @@ def test_rename_files(tmp_path):
     ).exists()
     assert (tmp_path / "foo/bar/baz.nii.gz").exists()
     assert (tmp_path / "foo/bar/sub-02_ses-M666_dwi_foo.nii.gz").exists()
+
+
+def test_get_b0_filter_error(tmp_path):
+    from clinica.utils.dwi import get_b0_filter
+
+    with pytest.raises(
+        FileNotFoundError,
+        match="File not found",
+    ):
+        get_b0_filter(tmp_path / "foo.bval")
+
+
+def test_get_b0_filter(tmp_path):
+    from clinica.utils.dwi import get_b0_filter
+
+    np.savetxt(tmp_path / "foo.bval", [1000, 1000, 0, 0, 0, 1000, 1000, 0])
+    assert_array_equal(get_b0_filter(tmp_path / "foo.bval"), np.array([2, 3, 4, 7]))
+    assert_array_equal(
+        get_b0_filter(tmp_path / "foo.bval", b_value_threshold=-1), np.array([])
+    )
+    assert_array_equal(
+        get_b0_filter(tmp_path / "foo.bval", b_value_threshold=500),
+        np.array([2, 3, 4, 7]),
+    )
+    assert_array_equal(
+        get_b0_filter(tmp_path / "foo.bval", b_value_threshold=1000), np.arange(8)
+    )
+    assert_array_equal(
+        get_b0_filter(tmp_path / "foo.bval", b_value_threshold=1001), np.arange(8)
+    )
+
+
+def test_count_b0s(tmp_path):
+    from clinica.utils.dwi import count_b0s
+
+    np.savetxt(tmp_path / "foo.bval", [1000, 1000, 0, 0, 0, 1000, 1000, 0])
+    assert count_b0s(tmp_path / "foo.bval") == 4
+    assert count_b0s(tmp_path / "foo.bval", b_value_threshold=-1) == 0
+    assert count_b0s(tmp_path / "foo.bval", b_value_threshold=500) == 4
+    assert count_b0s(tmp_path / "foo.bval", b_value_threshold=1000) == 8
+    assert count_b0s(tmp_path / "foo.bval", b_value_threshold=1001) == 8
+
+
+@pytest.mark.parametrize("extension", ["nii", "nii.gz"])
+def test_compute_average_b0(tmp_path, extension):
+    from clinica.utils.dwi import compute_average_b0
+
+    np.savetxt(tmp_path / "foo.bval", [1000, 1000, 0, 0, 0, 1000, 1000, 0])
+    img_data = np.zeros((5, 5, 5, 8))
+    img_data[2:4, 2:4, 2:4, 0:4] = 1.0
+    img_data[2:4, 2:4, 2:4, 4:8] = 2.0
+    img = nib.Nifti1Image(img_data, affine=np.eye(4))
+    nib.save(img, tmp_path / f"foo.{extension}")
+
+    # No filtering with the bvalues
+    out_file = compute_average_b0(tmp_path / f"foo.{extension}")
+    assert out_file == tmp_path / f"foo_avg_b0.{extension}"
+    result = nib.load(out_file)
+    assert result.shape == (5, 5, 5, 1)
+    expected = np.zeros((5, 5, 5, 1))
+    expected[2:4, 2:4, 2:4] = 1.5
+    assert_array_equal(result.get_fdata(), expected)
+
+    # With filtering
+    out_file = compute_average_b0(
+        tmp_path / f"foo.{extension}",
+        tmp_path / "foo.bval",
+    )
+    assert out_file == tmp_path / f"foo_avg_b0.{extension}"
+    result = nib.load(out_file)
+    assert result.shape == (5, 5, 5, 1)
+    expected = np.zeros((5, 5, 5, 1))
+    expected[2:4, 2:4, 2:4] = 1.5
+    assert_array_equal(result.get_fdata(), expected)
+
+
+@pytest.fixture
+def dwi_dataset(tmp_path):
```

Comment:
Might be dumb, but I don't understand what you are testing here.
Would instantiating a `DWIDataset` alone count as the test ?

Reply:
This isn't a test. All test functions start with the 'test' prefix. This is a pytest fixture that I'm defining to use this data in multiple tests.

Reply:
haaa okay i get it, thanks !